src/xregexp.js

```diff
@@ -51,7 +51,10 @@ var nativeTokens = {
     'class': /\\(?:[0-3][0-7]{0,2}|[4-7][0-7]?|x[\dA-Fa-f]{2}|u(?:[\dA-Fa-f]{4}|{[\dA-Fa-f]+})|c[A-Za-z]|[\s\S])|[\s\S]/
 };
 // Any backreference or dollar-prefixed character in replacement strings
-var replacementToken = /\$(?:{([\w$]+)}|(\d\d?|[\s\S]))/g;
+var replacementTokens = [
+    /\$(?:<([\w$]+)>)/g,
+    /\$(?:{([\w$]+)}|(\d\d?|[\s\S]))/g
+];
```

Comment:
I'd rather do this in one pass (rather than 2 or 3) for speed. But that aside, as written it doesn't make sense to do this in two passes since you're doing the same handling for `$2` in the `replacer` function even when it isn't relevant.

I think what you want here is to make `replacementToken` `/\$(?:{([\w$]+)}|<([\w$]+)>|(\d\d?|[\s\S]))/g`, then change `replacer ($0, $1, $2)` to `replacer($0, $1, $2, $3)` or something like `replacer($0, bracketed, angled, dollarToken)`, then put `bracketed = bracketed || angled` at the top of the `replacer` function.